File: Ide.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	// your code goes here
	int t,n,k,i,j,f,e;
	scanf("%d",&t);
	while(t--)
	{
	    scanf("%d %d",&n,&k);
	    int a[n+1][k];
	      for(i=0;i<n+1;i++)
	        for(j=0;j<k;j++)
    	    a[i][j]=0;
    	   
	    for(i=0;i<n;i++)
	    {   
	        scanf("%d",&f);
	        for(j=0;j<f;j++)
    	    {
    	        scanf("%d",&e);
    	        a[i][e-1]=1;
    	        a[n][e-1]++;
    	    }
	    }
	    for(i=0;i<=n;i++)
	       { for(j=0;j<k;j++)
    	    printf("%d",a[i][j]);
    	    printf("\n");
	       }
	   for(i=0;i<k;i++)
	   if(a[n][i]==0)
	   {
	       printf("sad\n");
	       return 0;
	   }
    	}
	
	return 0;
}

File: Ide.java
/* package whatever; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
class Ideone
{
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
		
		String s1="",s2="",s3,s4="",s5="";
		s3=br.readLine();
		int l1,l2,l3;
		l3=s3.length();
		int i;
		for( i=0;i<l3;i++)
		{
			char ch=s3.charAt(i);
			if(ch==' ')
			break;
			else
			s1=s1+ch;
		}
		s2=s3.substring(++i);
	    int t=Integer.parseInt(br.readLine());
	    	System.out.println(s1+" "+s2);
	    while(t!=0)
	    {
	        t--;
	        s3=br.readLine();
	       l2=s3.length();
		
		for( i=0;i<l2;i++)
		{
			char ch=s3.charAt(i);
			if(ch==' ')
			break;
			else
			s4=s4+ch;
		}
		s5=s3.substring(++i); 
		if(s1.equals(s4))
		{
		    s1=s5;
		    
		}
		else
		{
		    s2=s5;
		}
		System.out.println(s1+" "+s2);
	    }
	}
}